kucher/view/main_window/register_view_widget/import_export_dialog.py

```diff
@@ -0,0 +1,314 @@
+                    _logger.exception(f'Import failed: this parameter type is incorrect {reg_check}')
+                    return ErrorCodes.INCORRECT_TYPE, reg_check.name
+
+                elif len(imported_registers[reg_check.name]) != len(reg_check.cached_value):
+                    _logger.exception(f'Import failed: this parameter dimension is incorrect {reg_check}')
+                    return ErrorCodes.INCORRECT_DIMENSION, reg_check.name
+
+                elif reg_check.has_min_and_max_values and reg_check.type_id != Register.ValueType.BOOLEAN:
+                    if not (reg_check.min_value <= imported_registers[reg_check.name] <= reg_check.max_value):
+                        _logger.exception('Import failed: this parameter value is outside '
+                                          f'permitted range: {reg_check}')
+                        return ErrorCodes.OUTSIDE_RANGE, reg_check.name
+
+    except Exception as ex:
+        _logger.exception(f'Could not write registers: {str(ex)}')
+        return ErrorCodes.UNKNOWN, str(ex)
+
+    return ErrorCodes.NO_ERROR, ''
+
+
+def show_error_box(error_code, detail, file_name, parent):
+    if error_code == ErrorCodes.NOT_MUTABLE:
+        show_error('Import failed',
+                   f'Cannot import {file_name}',
+                   f'This parameter cannot be modified on this device:\n{detail}',
+                   parent)
+    elif error_code == ErrorCodes.INCORRECT_TYPE:
+        show_error('Import failed',
+                   f'Cannot import {file_name}',
+                   f'This parameter type is incorrect:\n{detail}',
+                   parent)
+    elif error_code == ErrorCodes.INCORRECT_DIMENSION:
+        show_error('Import failed',
+                   f'Cannot import {file_name}',
+                   f'This parameter dimension is incorrect:\n{detail}',
+                   parent)
+    elif error_code == ErrorCodes.OUTSIDE_RANGE:
+        show_error('Import failed',
+                   f'Cannot import {file_name}',
+                   f'This parameter value is outside permitted range:\n{detail}',
+                   parent)
+    else:
+        show_error('Import failed',
+                   f'Cannot import {file_name}',
+                   f'Reason: {detail}',
+                   parent)
+
+
+def write_registers(parent, file_name, registers, imported_registers):
+    async def executor():
+        unwritten_registers = []
+        for reg in imported_registers:
+            _attempt = 0
+            while True:
+                try:
+                    old_reg = next((r for r in registers if r.name == reg))
+                    await old_reg.write_through(imported_registers[reg])
+                    break
+
+                except Exception as ex:
+                    _attempt += 1
+                    _logger.exception(f'Register {reg} could not be loaded (attempt {_attempt}/3)')
+                    if _attempt == 3:
```

Comment:
Fragile code. Use `>=` instead; also, drop the read_through operation and use the cached value instead.